Typescript template tests are not working

## 🐛 Bug Report

As seen in output below, running npm test doesnt work. It looks similar to #303, but as it's error while importing tap there is not reason to be anything with autoload. 

**Setting "esModuleInterop" to true in tsconfig.json fix it.** 
I found that it's already fixed in #304, but still not merged.

## To Reproduce

Steps to reproduce the behavior:

```js
❯ fastify generate fastify-sandbox --lang=ts
generated .gitignore
generated tsconfig.json
generated test/helper.ts
generated src/app.ts
generated test/tsconfig.test.json
generated test/plugins/support.test.ts
generated test/routes/example.test.ts
generated src/plugins/README.md
generated src/routes/README.md
generated test/routes/root.test.ts
generated src/plugins/sensible.ts
generated src/plugins/support.ts
generated src/routes/root.ts
generated src/routes/example/index.ts
--> reading package.json in fastify-sandbox
edited package.json, saving
saved package.json
--> project fastify-sandbox generated successfully
run 'npm install' to install the dependencies
run 'npm start' to start the application
run 'npm build:ts' to compile the typescript application
run 'npm run dev' to start the application with pino-colada pretty logging (not suitable for production)
run 'npm test' to execute the unit tests
❯ cd fastify-sandbox
❯ pnpm i
Packages: +422
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
Packages are hard linked from the content-addressable store to the virtual store.
  Content-addressable store is at: /home/nezouse/.pnpm-store/v3
  Virtual store is at:             node_modules/.pnpm
Progress: resolved 423, reused 422, downloaded 1, added 422, done
 WARN  Failed to find "/fsevents/2.3.2" in lockfile during hoisting. Next aliases will not be hoisted: fsevents

dependencies:
+ fastify 3.12.0
+ fastify-autoload 3.5.2
+ fastify-cli 2.8.0
+ fastify-plugin 3.0.0
+ fastify-sensible 3.1.0

devDependencies:
+ @types/node 14.14.31
+ @types/tap 14.10.2
+ concurrently 6.0.0
+ fastify-tsconfig 1.0.1
+ tap 14.11.0
+ typescript 4.2.2
❯ pnpm test

> fastify-sandbox@1.0.0 test /home/nezouse/fastify-sandbox
> npm run build:ts && tsc -p test/tsconfig.test.json && tap test/**/*.test.ts


> fastify-sandbox@1.0.0 build:ts /home/nezouse/fastify-sandbox
> tsc

test/helper.ts:5:8 - error TS1259: Module '"/home/nezouse/fastify-sandbox/node_modules/.pnpm/@types/tap@14.10.2/node_modules/@types/tap/index"' can only be default-imported using the 'esModuleInterop' flag

5 import tap from 'tap';
         ~~~

  node_modules/.pnpm/@types/tap@14.10.2/node_modules/@types/tap/index.d.ts:13:1
    13 export = tap;
       ~~~~~~~~~~~~~
    This module is declared with using 'export =', and can only be used with a default import when using the 'esModuleInterop' flag.


Found 1 error.

 ERROR  Test failed. See above for more details.
```

## Expected behavior

```js
❯ npm test

> fastify-sandbox@1.0.0 test /home/nezouse/fastify-sandbox
> npm run build:ts && tsc -p test/tsconfig.test.json && tap test/**/*.test.ts


> fastify-sandbox@1.0.0 build:ts /home/nezouse/fastify-sandbox
> tsc

 PASS  test/plugins/support.test.ts 1 OK 48.572ms
 PASS  test/routes/example.test.ts 1 OK 2s
 PASS  test/routes/root.test.ts 1 OK 2s

                         
  🌈 SUMMARY RESULTS 🌈  
                         

Suites:   3 passed, 3 of 3 completed
Asserts:  3 passed, of 3
Time:     8s
--------------------|----------|----------|----------|----------|-------------------|
File                |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
--------------------|----------|----------|----------|----------|-------------------|
All files           |      100 |      100 |      100 |      100 |                   |
 src                |      100 |      100 |      100 |      100 |                   |
  app.ts            |      100 |      100 |      100 |      100 |                   |
 src/plugins        |      100 |      100 |      100 |      100 |                   |
  sensible.ts       |      100 |      100 |      100 |      100 |                   |
  support.ts        |      100 |      100 |      100 |      100 |                   |
 src/routes         |      100 |      100 |      100 |      100 |                   |
  root.ts           |      100 |      100 |      100 |      100 |                   |
 src/routes/example |      100 |      100 |      100 |      100 |                   |
  index.ts          |      100 |      100 |      100 |      100 |                   |
--------------------|----------|----------|----------|----------|-------------------|
```

## Your Environment

- *node version*: v14.15.5
- *fastify version*: template installed as today, 
```js
{
    "fastify": "^3.0.0",
    "fastify-cli": "^2.8.0"
}
```
- *os*: Linux on WSL

--- a/generate.js
+++ b/generate.js
@@ -45,7 +45,7 @@ const typescriptTemplate = {
   dir: 'app-ts',
   main: 'app.ts',
   scripts: {
-    test: 'npm run build:ts && tsc -p test/tsconfig.test.json && tap test/**/*.test.ts',
+    test: 'tap test/**/*.test.ts',
     start: 'npm run build:ts && fastify start -l info dist/app.js',
     'build:ts': 'tsc',
     dev: 'tsc && concurrently -k -p "[{name}]" -n "TypeScript,App" -c "yellow.bold,cyan.bold"  "tsc -w" "fastify start -w -l info -P dist/app.js"'

--- a/generate.js
+++ b/generate.js
@@ -17,7 +17,7 @@ const javascriptTemplate = {
   dir: 'app',
   main: 'app.js',
   scripts: {
-    test: 'tap test/**/*.test.js',
+    test: 'tap "test/**/*.test.js"',
     start: 'fastify start -l info app.js',
     dev: 'fastify start -w -l info -P app.js'
   },
@@ -45,7 +45,7 @@ const typescriptTemplate = {
   dir: 'app-ts',
   main: 'app.ts',
   scripts: {
-    test: 'npm run build:ts && tap test/**/*.test.ts',
+    test: 'npm run build:ts && tap "test/**/*.test.ts"',
     start: 'npm run build:ts && fastify start -l info dist/app.js',
     'build:ts': 'tsc',
     dev: 'tsc && concurrently -k -p "[{name}]" -n "TypeScript,App" -c "yellow.bold,cyan.bold"  "tsc -w" "fastify start -w -l info -P dist/app.js"'

--- a/generate.js
+++ b/generate.js
@@ -45,7 +45,7 @@ const typescriptTemplate = {
   dir: 'app-ts',
   main: 'app.ts',
   scripts: {
-    test: 'tap test/**/*.test.ts',
+    test: 'npm run build:ts && tap test/**/*.test.ts',
     start: 'npm run build:ts && fastify start -l info dist/app.js',
     'build:ts': 'tsc',
     dev: 'tsc && concurrently -k -p "[{name}]" -n "TypeScript,App" -c "yellow.bold,cyan.bold"  "tsc -w" "fastify start -w -l info -P dist/app.js"'

--- a/templates/app-ts/__taprc
+++ b/templates/app-ts/__taprc
@@ -1,0 +1,3 @@
+test-env: [
+  TS_NODE_PROJECT=./test/tsconfig.test.json
+]

--- a/templates/app-ts/test/helper.ts
+++ b/templates/app-ts/test/helper.ts
@@ -2,7 +2,7 @@
 import Fastify from 'fastify'
 import fp from 'fastify-plugin'
 import App from '../src/app'
-import tap from 'tap';
+import * as tap from 'tap';
 
 export type Test = typeof tap['Test']['prototype'];
 

--- a/test/generate-typescript.test.js
+++ b/test/generate-typescript.test.js
@@ -129,7 +129,7 @@ function define (t) {
         // by default this will be ISC but since we have a MIT licensed pkg file in upper dir, npm will set the license to MIT in this case
         // so for local tests we need to accept MIT as well
         t.ok(pkg.license === 'ISC' || pkg.license === 'MIT')
-        t.equal(pkg.scripts.test, 'npm run build:ts && tap test/**/*.test.ts')
+        t.equal(pkg.scripts.test, 'npm run build:ts && tap "test/**/*.test.ts"')
         t.equal(pkg.scripts.start, 'npm run build:ts && fastify start -l info dist/app.js')
         t.equal(pkg.scripts['build:ts'], 'tsc')
         t.equal(pkg.scripts.dev, 'tsc && concurrently -k -p "[{name}]" -n "TypeScript,App" -c "yellow.bold,cyan.bold"  "tsc -w" "fastify start -w -l info -P dist/app.js"')
@@ -143,7 +143,7 @@ function define (t) {
         t.equal(pkg.devDependencies.tap, cliPkg.devDependencies.tap)
         t.equal(pkg.devDependencies.typescript, cliPkg.devDependencies.typescript)
 
-        const testGlob = pkg.scripts.test.split(' ')[5]
+        const testGlob = pkg.scripts.test.split(' ')[5].replace(/"/g, '')
 
         t.equal(minimatch.match(['test/routes/plugins/more/test/here/ok.test.ts'], testGlob).length, 1)
         resolve()

--- a/test/generate-typescript.test.js
+++ b/test/generate-typescript.test.js
@@ -103,7 +103,7 @@ function define (t) {
     })
   })
 
-  test('should finish succesfully with typescript template', async (t) => {
+  test('should finish successfully with typescript template', async (t) => {
     t.plan(22 + Object.keys(expected).length)
     try {
       await generate(workdir, typescriptTemplate)
@@ -129,7 +129,7 @@ function define (t) {
         // by default this will be ISC but since we have a MIT licensed pkg file in upper dir, npm will set the license to MIT in this case
         // so for local tests we need to accept MIT as well
         t.ok(pkg.license === 'ISC' || pkg.license === 'MIT')
-        t.equal(pkg.scripts.test, 'npm run build:ts && tsc -p test/tsconfig.test.json && tap test/**/*.test.ts')
+        t.equal(pkg.scripts.test, 'tap test/**/*.test.ts')
         t.equal(pkg.scripts.start, 'npm run build:ts && fastify start -l info dist/app.js')
         t.equal(pkg.scripts['build:ts'], 'tsc')
         t.equal(pkg.scripts.dev, 'tsc && concurrently -k -p "[{name}]" -n "TypeScript,App" -c "yellow.bold,cyan.bold"  "tsc -w" "fastify start -w -l info -P dist/app.js"')
@@ -143,7 +143,7 @@ function define (t) {
         t.equal(pkg.devDependencies.tap, cliPkg.devDependencies.tap)
         t.equal(pkg.devDependencies.typescript, cliPkg.devDependencies.typescript)
 
-        const testGlob = pkg.scripts.test.split(' ')[9]
+        const testGlob = pkg.scripts.test.split(' ')[1]
 
         t.equal(minimatch.match(['test/routes/plugins/more/test/here/ok.test.ts'], testGlob).length, 1)
         resolve()

--- a/test/generate-typescript.test.js
+++ b/test/generate-typescript.test.js
@@ -129,7 +129,7 @@ function define (t) {
         // by default this will be ISC but since we have a MIT licensed pkg file in upper dir, npm will set the license to MIT in this case
         // so for local tests we need to accept MIT as well
         t.ok(pkg.license === 'ISC' || pkg.license === 'MIT')
-        t.equal(pkg.scripts.test, 'tap test/**/*.test.ts')
+        t.equal(pkg.scripts.test, 'npm run build:ts && tap test/**/*.test.ts')
         t.equal(pkg.scripts.start, 'npm run build:ts && fastify start -l info dist/app.js')
         t.equal(pkg.scripts['build:ts'], 'tsc')
         t.equal(pkg.scripts.dev, 'tsc && concurrently -k -p "[{name}]" -n "TypeScript,App" -c "yellow.bold,cyan.bold"  "tsc -w" "fastify start -w -l info -P dist/app.js"')
@@ -143,7 +143,7 @@ function define (t) {
         t.equal(pkg.devDependencies.tap, cliPkg.devDependencies.tap)
         t.equal(pkg.devDependencies.typescript, cliPkg.devDependencies.typescript)
 
-        const testGlob = pkg.scripts.test.split(' ')[1]
+        const testGlob = pkg.scripts.test.split(' ')[5]
 
         t.equal(minimatch.match(['test/routes/plugins/more/test/here/ok.test.ts'], testGlob).length, 1)
         resolve()

--- a/test/generate.test.js
+++ b/test/generate.test.js
@@ -146,7 +146,7 @@ function define (t) {
         // by default this will be ISC but since we have a MIT licensed pkg file in upper dir, npm will set the license to MIT in this case
         // so for local tests we need to accept MIT as well
         t.ok(pkg.license === 'ISC' || pkg.license === 'MIT')
-        t.equal(pkg.scripts.test, 'tap test/**/*.test.js')
+        t.equal(pkg.scripts.test, 'tap "test/**/*.test.js"')
         t.equal(pkg.scripts.start, 'fastify start -l info app.js')
         t.equal(pkg.scripts.dev, 'fastify start -w -l info -P app.js')
         t.equal(pkg.dependencies['fastify-cli'], '^' + cliPkg.version)
@@ -156,7 +156,7 @@ function define (t) {
         t.equal(pkg.dependencies['fastify-sensible'], cliPkg.devDependencies['fastify-sensible'])
         t.equal(pkg.devDependencies.tap, cliPkg.devDependencies.tap)
 
-        const testGlob = pkg.scripts.test.split(' ')[1]
+        const testGlob = pkg.scripts.test.split(' ')[1].replace(/"/g, '')
         t.equal(minimatch.match(['test/services/plugins/more/test/here/ok.test.js'], testGlob).length, 1)
         resolve()
       })

